torch/csrc/autograd/python_saved_variable_hooks.cpp

```diff
@@ -9,19 +11,35 @@ namespace torch { namespace autograd {
     unpack_hook_(unpack_hook.release().ptr()) {}
 
   void PySavedVariableHooks::call_pack_hook(at::Tensor &tensor) {
-    TORCH_CHECK_NOT_IMPLEMENTED(false, "Hooks are not implemented yet");
+    py::gil_scoped_acquire acquire;
+    auto pack_hook = py::reinterpret_borrow<py::function>(pack_hook_); // borrows the reference to the hook
```

Comment:
Do you plan to leave all the ref counting comments?

Reply:
I tried simplifying the ref counting comments